File: airbyte-integrations/connectors/destination-azure-blob-storage/src/main/java/io/airbyte/integrations/destination/azure_blob_storage/writer/AzureBlobStorageWriter.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.azure_blob_storage.writer;

import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
import java.io.IOException;
import java.util.UUID;

/**
 * {@link AzureBlobStorageWriter} is responsible for writing Airbyte stream data to an
 * AzureBlobStorage location in a specific format.
 */
public interface AzureBlobStorageWriter {

  /**
   * Write an Airbyte record message to an AzureBlobStorage object.
   */
  void write(UUID id, AirbyteRecordMessage recordMessage) throws IOException;

  /**
   * Close the AzureBlobStorage writer for the stream.
   */
  void close(boolean hasFailed) throws IOException;

}

File: airbyte-integrations/connectors/destination-azure-blob-storage/src/main/kotlin/io/airbyte/integrations/destination/azure_blob_storage/AzureBlobStorageWriter.kt
/*
 * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.azure_blob_storage

import io.airbyte.cdk.load.command.DestinationStream
import io.airbyte.cdk.load.file.azureBlobStorage.AzureBlob
import io.airbyte.cdk.load.write.DestinationWriter
import io.airbyte.cdk.load.write.StreamLoader
import io.airbyte.cdk.load.write.object_storage.ObjectStorageStreamLoaderFactory
import javax.inject.Singleton

@Singleton
class AzureBlobStorageWriter(
    private val streamLoaderFactory: ObjectStorageStreamLoaderFactory<AzureBlob, *>,
) : DestinationWriter {
    override fun createStreamLoader(stream: DestinationStream): StreamLoader {
        return streamLoaderFactory.create(stream)
    }
}
